frame/staking/src/lib.rs

```diff
@@ -2299,11 +2299,21 @@ impl<T: Trait> Module<T> {
 		Self::bonded(stash).and_then(Self::ledger).map(|l| l.active).unwrap_or_default()
 	}
 
-	/// internal impl of [`slashable_balance_of`] that returns [`VoteWeight`].
-	fn slashable_balance_of_vote_weight(stash: &T::AccountId) -> VoteWeight {
-		<T::CurrencyToVote as Convert<BalanceOf<T>, VoteWeight>>::convert(
-			Self::slashable_balance_of(stash)
-		)
+	/// Internal impl of [`slashable_balance_of`] that returns [`VoteWeight`].
+	fn slashable_balance_of_vote_weight(stash: &T::AccountId, issuance: BalanceOf<T>) -> VoteWeight {
+		T::CurrencyToVote::to_vote(Self::slashable_balance_of(stash), issuance)
+	}
+
+	/// Returns a closure around `slashable_balance_of_vote_weight` that can be passed around.
+	///
+	/// This prevents call sites from repeatedly requesting `total_issuance` from backend.
+	pub(crate) fn weight_of_fn() -> Box<dyn Fn(&T::AccountId) -> VoteWeight> {
+		// TODO: changing this to unboxed `impl Fn(..)` return type and the module will still
+		// compile, while some types in mock fail to resolve, all in all, wtf is all I can say.
+		let issuance = T::Currency::total_issuance();
+		Box::new(move |who: &T::AccountId| -> VoteWeight {
+			Self::slashable_balance_of_vote_weight(who, issuance)
```

Comment:
Are we sure that `total_issuance` won't change in between the calls? This seems like a subtlety that might cause some errors in the future.

Reply:
Indeed a correct note, but as far as I can reason about, it is safe. 

In the context of staking, there are two places where we do election (and thus we use total issuance and this function):

1. on-chain election. In this case _nothing_ else happens in between on chain while we are executing the election and thus safe (in principle this can change maybe when we add threads to the runtime).
2. off-chain election. In this case, it is theoretically possible for the total issuance to change. While the OCW is running the chain can transition into a new state with a new total issuance, but I see that as such an unlikely scenario that I decided to skip it. 

It should do a check and see, if for example the total issuance happens to change a little, what are the consequences.

Reply:
> off-chain election. In this case, it is theoretically possible for the total issuance to change. While the OCW is running the chain can transition into a new state with a new total issuance, but I see that as such an unlikely scenario that I decided to skip it.

That's not the case, off-chain worker runs within a pinned state, so new blocks don't affect it.

Reply:
Thanks for clarifying, then we are done.